src/python/gudhi/bottleneck.cc

```diff
@@ -12,15 +12,18 @@
 
 #include <pybind11_diagram_utils.h>
 
+// Indices are added internally in bottleneck_distance, they are not needed in the input.
+static auto make_point(double x, double y, py::ssize_t) { return std::pair(x, y); };
+
 // For compatibility with older versions, we want to support e=None.
 // In C++17, the recommended way is std::optional<double>.
```

Comment:
As we are now in C++17, it may be a good opportunity to fix this, no ? (not mandatory on this PR, we can also open an issue)

Reply:
It isn't broken, optional is just another way to handle this. But ok, done.